### PR TITLE
Make sure duplicates work better

### PR DESCRIPTION
**Pull request by libre-man**

### CODE REVIEW

File: codegra_fs/cgfs.py [hunk 1]
```diff
@@ -1416,8 +1416,25 @@ def strippath(self, path: str) -> str:
...
         assert cgapi is not None
 
-        for course in cgapi.get_courses():
+        courses = cgapi.get_courses()
+        for dups in codegra_fs.utils.find_all_dups(
+            courses, lambda x: x['name']
+        ):
+            end = 4
+            if len(dups) != len(set(d['created_at'][:4] for d in dups)):
+                end = None
```
Comment (olmokramer) [inline codegra_fs/cgfs.py]: Maybe we should still remove the time portion of the name? `end = 10`?

File: codegra_fs/cgfs.py [hunk 2]
```diff
@@ -1416,8 +1416,25 @@ def strippath(self, path: str) -> str:
...
+                date = dup['created_at'][:end]
+                dup['name'] += ' - ' + date
+
+        for course in courses:
             assignments = course['assignments']
+            for dups in codegra_fs.utils.find_all_dups(
+                assignments, lambda x: x['name']
+            ):
+                for dup in dups:
+                    dup['name'] += ' - ' + dup['created_at']
```
Comment (olmokramer) [inline codegra_fs/cgfs.py]: Same as above. Should we also do that for submissions in an assignment? Probably not remove the hours and minutes, maybe not even the seconds, but everything after that can be removed IMO.

File: codegra_fs/cgfs.py (hunk 1 above)
Comment (libre-man) [reply codegra_fs/cgfs.py]: Yeah that is probably a good idea.

File: codegra_fs/cgfs.py (hunk 2 above)
Comment (libre-man) [reply codegra_fs/cgfs.py]: Agree, the seconds are also useless imo.

File: codegra_fs/cgfs.py (hunk 1 above)
Comment (libre-man) [reply codegra_fs/cgfs.py]: No that is not possible here as we detected that the created date of these duplicates is very close to each other. We should do this however for other cases.

**Pull request merged**